[Cppcheck] Support the option `-I` (#476)
sider/sider-docs: docs/getting-started/custom-configuration.md

When the option is `false`, Sider will skip these installation steps and analyze with the pre-installed default version.
You might want to set the option to `false` if you don't configure analyzer and don't want to see warnings.

## `linter.<analyzer_id>.apt`

_Type:_ `string`, `string[]`

> Some projects particularly written in C/C++ depend on development packages. Sider can install packages with the `apt` package manager, which is on `Debian` based Linux distributions.

Reviewer: ```suggestion
Development packages provided by the OS environment may be necessary, particularly for projects written in C/C++. Sider lets you install packages with APT, which is a package manager for `Debian` based Linux distributions.
```

Author: Agreed. commit 68b002999a1ee1fb86b7471e74738d0b9c6d54d8

The `apt` option allows you to specify a list of development packages your project depends on.
The packages must satisfy the conditions below:

> - Packages compatible with [our Docker image](https://github.com/sider/devon_rex/blob/master/base/Dockerfile).
> - Packages with "-dev" suffix in names are available.
> - A specific version number can be requested with the `<name>=<version>` format.

Reviewer: How about this? I think these conditions *must* be applied for the `apt` options.

```suggestion
- Packages must be compatible with [our Docker image](https://github.com/sider/devon_rex/blob/master/base/Dockerfile).
- Package names must be suffixed with "-dev".
- Each package name must be formatted as `<name>` or `<name>=<version>`.
```

Author: Agreed. commit 68b002999a1ee1fb86b7471e74738d0b9c6d54d8

Below is an example of how you install the latest version of `libgdbm-dev` and the specific version (0.99.8-2) of `libfastjson-dev`.

```yaml
linter:
  clang_tidy:
    apt:
      - libgdbm-dev
      - libfastjson-dev=0.99.8-2
```

## `linter.<analyzer_id>.include-path`

_Type:_ `string`, `string[]`

> Some C/C++ analyzers can handle include paths like any other C/C++ preprocessor. With Sider, the `include-path` option allows you to add directories to include search path.

Reviewer: I thought *analyzers* and *preprocessors* are not in the same category, so I removed "any other."

```suggestion
Some C/C++ analyzers can handle include paths like C/C++ preprocessors can. With Sider, the `include-path` option allows you to add directories to include search paths.
```

Author: Agreed. commit 68b002999a1ee1fb86b7471e74738d0b9c6d54d8

For example:

```yaml
linter:
  clang_tidy:
    include-path:
      - myinclude
      - foo/include
      - /usr/include/libfastjson
```

> Sider treats this option as a compilation option `-I` and passes it to the `clang-tidy` command internally as:

Reviewer 2: [suggestion] Users may be confused for `clang-tidy` if they want to use `cppcheck`. How about rewriting it as follows? 👇 

```suggestion
Sider treats this option as a compilation option `-I` and passes it to the analysis tool command internally as:
```

Author: Agreed. But I will adopt another suggestion at https://github.com/sider/sider-docs/pull/476#discussion_r457905229

> Sider treats this option as a compilation option `-I` and passes it to the `clang-tidy` command internally as:

Reviewer: This section is not specific for Clang-Tidy, so I edited here like this:

```suggestion
Sider treats this option as a compilation option `-I` and passes it as command-line arguments internally. For example, [Clang-Tidy](../tools/cplusplus/clang-tidy.md) treats this option like this:
```

Author: Agreed. commit 68b002999a1ee1fb86b7471e74738d0b9c6d54d8
